lookin/__init__.py

```diff
@@ -1,93 +1,69 @@
+from homeassistant.core import HomeAssistant
 from homeassistant.exceptions import ConfigEntryNotReady
 from homeassistant.helpers.aiohttp_client import async_get_clientsession
-from homeassistant.helpers.entity import Entity
-
-from .const import DEVICES, DOMAIN, LOGGER, PLATFORMS, PROTOCOL
-from .error import DeviceNotFound
-from .protocol import LookInHttpProtocol  # TODO: move protocol into a PyPI package
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
 
-if TYPE_CHECKING:
-    from homeassistant.config_entries import ConfigEntry
-    from homeassistant.core import HomeAssistant
+from .aiolookin import (
+    DeviceNotFound,
+    LookInHttpProtocol,
+    LookinUDPSubscriptions,
+    start_lookin_udp,
+)
+from .const import DOMAIN, PLATFORMS
+from .models import LookinData
 
+LOGGER = logging.getLogger(__name__)
 
-async def async_setup_entry(hass: "HomeAssistant", entry: "ConfigEntry") -> bool:
-    LOGGER.warning("Lookin service started")
-    LOGGER.warning("config_entry.data - <%s>", entry.data)
 
-    LOGGER.warning("Lookin service CONF_DEVICE_ID <%s>", entry.data[CONF_DEVICE_ID])
-    LOGGER.warning("Lookin service CONF_HOST <%s>", entry.data[CONF_HOST])
-
-    LOGGER.warning("Lookin service entry.entry_id <%s>", entry.entry_id)
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up lookin from a config entry."""
 
+    host = entry.data[CONF_HOST]
     lookin_protocol = LookInHttpProtocol(
-        host=entry.data[CONF_HOST], session=async_get_clientsession(hass)
+        host=host, session=async_get_clientsession(hass)
     )
 
     try:
+        lookin_device = await lookin_protocol.get_info()
         devices = await lookin_protocol.get_devices()
     except DeviceNotFound as ex:
         raise ConfigEntryNotReady from ex
 
-    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {
-        CONF_HOST: entry.data[CONF_HOST],
-        CONF_DEVICE_ID: entry.data[CONF_DEVICE_ID],
-        CONF_NAME: entry.data[CONF_NAME],
-        DEVICES: devices,
-        PROTOCOL: lookin_protocol,
-    }
-
-    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
-
-    return True
-
-
-class LookinSensor(Entity):
-    def __init__(
-        self, device_id: str, device_name: str, unit_of_measurement: str
-    ) -> None:
-        self.current_value: Optional[Any] = None
-        self._device_id = device_id
-        self._name = f"{DOMAIN}.{device_name}.{self.device_class}"
-        self._unique_id = f"{device_id}-{self.device_class}"
-        self._unit_of_measurement = unit_of_measurement
-        self._is_available = True
-        self._firmware: Optional[str] = None
-
-    @property
-    def name(self) -> str:
-        return self._name
+    meteo_coordinator: DataUpdateCoordinator = DataUpdateCoordinator(
+        hass,
+        LOGGER,
+        name=entry.title,
+        update_method=lookin_protocol.get_meteo_sensor,
+        update_interval=timedelta(seconds=15),
```

Comment:
```suggestion
        update_interval=timedelta(minutes=5),
```

The temp and humidity aren't going to rapidly change so probably 5 minutes is ok here.